android/engine/src/main/java/org/smartregister/fhircore/engine/util/ParallelUtil.kt

```diff
@@ -34,10 +37,16 @@ suspend fun <A, B> Iterable<A>.pmap(f: suspend (A) -> B): Iterable<B> = coroutin
 }
 
 /**
- * Launch a new coroutine for each loop iteration using async.
+ * Launch a new coroutine for each loop iteration using launch and the Default Dispatcher for
+ * computationaly intensive tasks.
  *
  * @param T the type of elements in the iterable
  */
 suspend fun <T> Iterable<T>.forEachAsync(action: suspend (T) -> Unit): Unit = coroutineScope {
-  forEach { async { action(it) } }
+  forEach { launch(Dispatchers.Default) { action(it) } }
```

Comment:
Launching coroutines in a for each would negatively impact the app's performance.

Reply:
This is because of the overhead incurred when launching a coroutine.

Reply:
I suppose it depends on the intended usage for the async for each. _e.g. if we are going to make a network request for each item in the foreach then the overhead will be negligible compared?_